### DISCUSSION TITLE
[postcss] how to split Css files for each route

### DISCUSSION THREAD

**Discussion by ahakem** [vercel/next.js#11280] [Help]
Hey,
I'm using `postcss`  and trying to bootstrap the project,

I can see that all css files are added only via the map `_app.js`  which will include all styles all over the app in single .css file which may impact the performance.

what I want to do here is to have css file loaded for each page to avoid unused css.
how can I do that ?

Thank you

**Comment by timneutkens** [comment]
Using CSS modules, these will be automatically split per-page: https://nextjs.org/docs/basic-features/built-in-css-support#adding-component-level-css

**Reply by ahakem** [reply]
Thank you, but what if I want to use the power of `postcss`  plugins ?
I believe it may webpack configration but not sure who to achieve that

**Reply by Timer** [reply]
Next.js allows you to configure PostCSS using `postcss.config.js(on)`:
https://nextjs.org/docs/advanced-features/customizing-postcss-config

**Reply by ahakem** [reply]
Thank again but `postcss.config.js` is to configure the plugin used in postcss, not how to serve the output files for the application. @Timer

**Reply by Timer** [reply]
Next.js automatically splits CSS files by page. You do not need to configure anything to get this behavior.

**Reply by ahakem** [reply]
@timer, 
I've checked that, I have 2 pages,  which are `index` and `/home` 
I made a build and chked the served files, and both pages using same file `http://localhost:5000/_next/static/css/7e9063fa5d72aacfd41e.css`

**Reply by Timer** [reply]
Are you using `@zeit/next-css` or `@zeit/next-sass`?

**Reply by ahakem** [reply]
`@zeit/next-css`

**Reply by Timer** [reply]
`@zeit/next-css` does not support CSS splitting. Please delete all usage of `@zeit/next-css` and use Next.js' built-in CSS support.

**Reply by ahakem** [reply]
Thank you, I'll check that.

**Reply by mataqque** [reply]
doesn't work for me, remove next-css and next-sass, use the support that next.js provides https://nextjs.org/docs/basic-features/built-in-css-support#sass-support. and it doesn't work I have all the css in one file again.
the only thing I can think of is to use webpack and do it manually.

**Reply by ijakparov** [reply]
> doesn't work for me, remove next-css and next-sass, use the support that next.js provides https://nextjs.org/docs/basic-features/built-in-css-support#sass-support. and it doesn't work I have all the css in one file again.
> the only thing I can think of is to use webpack and do it manually.

Could you share your result?
